### PR TITLE
Lcel workshop 2024

### PR DESCRIPTION
**Pull request by TLongP**
The scripts for dida workshop 2024, **LangChain Expression Language for building LLM production pipelines**.

### CODE REVIEW

File: langchain-workshop/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+To use `langchain-workshop/workshop-lcel.ipynb` and `langchain-workshop/agent_example.ipynb` you need to do the following:
+
+# Download the models
+Run ```download_weights.ipynb``` to donwload ```BAAI/bge-large-en-v1.5``` and ```BAAI/bge-reranker-large```.
```
Comment (astoff) [inline langchain-workshop/README.md]:
```suggestion
Run ```download_weights.ipynb``` to download ```BAAI/bge-large-en-v1.5``` and ```BAAI/bge-reranker-large```.

```

File: langchain-workshop/download_weights.ipynb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "d:\\projects\\dida-langchain-workshop\\.venv\\Lib\\site-packages\\huggingface_hub\\file_download.py:1194: UserWarning: `local_dir_use_symlinks` parameter is deprecated and will be ignored. The process to download files to a local folder has been updated and do not rely on symlinks anymore. You only need to pass a destination folder as`local_dir`.\n",
```
Comment (astoff) [inline langchain-workshop/download_weights.ipynb]: Did you keep the output cells on purpose?

File: langchain-workshop/llm-papers/2305.18290v2.Direct_Preference_Optimization__Your_Language_Model_is_Secretly_a_Reward_Model.pdf
Comment (astoff) [inline langchain-workshop/llm-papers/2305.18290v2.Direct_Preference_Optimization__Your_Language_Model_is_Secretly_a_Reward_Model.pdf:1]: Why not just provide a link to the paper on arXiv?

File: langchain-workshop/README.md (hunk 1 above)
Comment (TLongP) [reply langchain-workshop/README.md]: fixed.

File: langchain-workshop/download_weights.ipynb (hunk 2 above)
Comment (TLongP) [reply langchain-workshop/download_weights.ipynb]: Fixed.

File: langchain-workshop/llm-papers/2305.18290v2.Direct_Preference_Optimization__Your_Language_Model_is_Secretly_a_Reward_Model.pdf
Comment (TLongP) [reply langchain-workshop/llm-papers/2305.18290v2.Direct_Preference_Optimization__Your_Language_Model_is_Secretly_a_Reward_Model.pdf:1]: I changed it. Now you can load the pdfs directly or you can mannually download them.

**Comment by TLongP** [comment]
I make also a small change to make the python file into a notebook.

**Review by astoff — APPROVED** [review]

**Pull request merged**